buildSrc/src/main/kotlin/conventions/testArtifacts.gradle.kts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package conventions
```

Comment:
A convention to create test artifact jars. This provides support for test resources / classes to be consumed across projects using the kotlin dsl.

Previously, we'd use the following gradle syntax:

```
    testImplementation project(':bson').sourceSets.test.output
    testImplementation project(':driver-core').sourceSets.test.output
    testImplementation project(':driver-sync').sourceSets.test.output
```

This isn't supported in the kotlin dsl. Infact `sourceSets` is even greyed out in intellij with the groovy. So its unclear how this is resolved.

There are test fixtures in gradle now - but I'm not sure they meet our requirements (and require their own project layout conventions).